homeassistant/components/sensor/wunderground.py

```diff
@@ -629,32 +627,31 @@ def _get_attributes(rest):
     vol.Inclusive(CONF_LONGITUDE, 'coordinates',
                   'Latitude and longitude must exist together'): cv.longitude,
     vol.Required(CONF_MONITORED_CONDITIONS):
-        vol.All(cv.ensure_list, vol.Length(min=1), [vol.In(SENSOR_TYPES)]),
-    vol.Optional(CONF_ENTITY_NAMESPACE,
-                 default=DEFAULT_ENTITY_NAMESPACE): cv.string,
+        vol.All(cv.ensure_list, vol.Length(min=1), [vol.In(SENSOR_TYPES)])
 })
 
-# Stores a list of entity ids we added in order to support multiple stations
-# at once.
-ADDED_ENTITY_IDS_KEY = 'wunderground_added_entity_ids'
-
 
 @asyncio.coroutine
 def async_setup_platform(hass: HomeAssistantType, config: ConfigType,
                          async_add_devices, discovery_info=None):
     """Set up the WUnderground sensor."""
-    hass.data.setdefault(ADDED_ENTITY_IDS_KEY, set())
-
     latitude = config.get(CONF_LATITUDE, hass.config.latitude)
     longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
-    namespace = config.get(CONF_ENTITY_NAMESPACE)
+    pws_id = config.get(CONF_PWS_ID)
 
     rest = WUndergroundData(
-        hass, config.get(CONF_API_KEY), config.get(CONF_PWS_ID),
+        hass, config.get(CONF_API_KEY), pws_id,
         config.get(CONF_LANG), latitude, longitude)
+
+    if pws_id is None:
+        unique_id_base = "@{},{}".format(longitude, latitude)
+    else:
+        # Manually specified weather station, use that for unique_id
+        unique_id_base = pws_id
```

Comment:
Unique IDs should not be manually configurable.

Reply:
I don't completely understand this comment. In the configuration for `wunderground` there are two ways to select which data is fetched from WUnderground: `lat/lon` (the coordinates of your home) **or** `pws_id` (the weather station id you want to pull data from).

So the unique ids are generated based on those values - therefore the unique ids are not really manually configurable. (Only in the way other unique ids are also configurable, for example in the cast integration `host: 192.168.178.42` will have a different unique id from `host: 123.45.67.89`)

Reply:
I wasn't sure how configurable the pws_id was. Might be ok in this case.

Ip addresses should not be used for unique ids, as they are not unique and not persistent (always).

Reply:
weather station should be fine for unique id. Note that IP addresses should never be used as unique id.